src/TicketValidator.java

```diff
@@ -17,57 +16,81 @@ public TicketValidator() {
     public void validateTickets(List<BusTicket> tickets) {
         for (BusTicket ticket : tickets) {
             totalTickets++;
-            List<String> violations = validateTicket(ticket);
-            if (violations.isEmpty()) {
+            try {
+                validateTicket(ticket);
                 validTickets++;
-            } else {
-                for (String violation : violations) {
-                    violationsCount.put(violation, violationsCount.getOrDefault(violation, 0) + 1);
-                }
-                System.err.println("Ticket validation error: " + violations + " for ticket class: " + ticket.ticketClass);
+            } catch (InvalidTicketException e) {
+                System.err.println(e.getMessage() + " for ticket class: " + ticket.getTicketClass());
             }
         }
     }
 
-    private List<String> validateTicket(BusTicket ticket) {
+    private void validateTicket(BusTicket ticket) throws InvalidTicketException {
```

Comment:
why have you changed return type?

Reply:
return type was changed to void because the method purpose is to validate the ticket and throw an exception if something's wrong rather than returning a list of violations